quantum/process_keycode/process_autocorrect.c

```diff
@@ -25,65 +22,65 @@
  * @return true Continue processing keycodes, and send to host
  * @return false Stop processing keycodes, and don't send to host
  */
-bool process_autocorrection(uint16_t keycode, keyrecord_t* record) {
-    static uint8_t typo_buffer[AUTOCORRECTION_MAX_LENGTH] = {KC_SPC};
-    static uint8_t typo_buffer_size                       = 1;
 
-    if (keycode == AUTO_CTN) {
+bool process_autocorrect(uint16_t keycode, keyrecord_t* record) {
+    static uint8_t typo_buffer[AUTOCORRECT_MAX_LENGTH] = {KC_SPC};
+    static uint8_t typo_buffer_size                    = 1;
+
+    if (keycode == AUTOCRT) {
         if (record->event.pressed) {
             typo_buffer_size = 0;
-            userspace_config.autocorrection ^= 1;
-            eeconfig_update_user(userspace_config.raw);
+            keymap_config.autocorrect_enable ^= 1;
+            eeconfig_update_keymap(keymap_config.raw);
         }
         return false;
     }
 
-    if (!userspace_config.autocorrection) {
+    if (!keymap_config.autocorrect_enable) {
         typo_buffer_size = 0;
         return true;
     }
 
+    if (!record->event.pressed) {
+        return true;
+    }
+
     switch (keycode) {
         case KC_LSFT:
```

Comment:
Minor suggestion: add `case KC_CAPS:` here so that Caps Lock presses are ignored as well.